docs/codebase-notes/messages-reducer.md

```diff
@@ -0,0 +1,262 @@
+# Messages Reducer Notes
```

Comment:
wonder if this should be closer to the Web Console codebase. Is this going to go stale...

Reply:
We don't have any kind of a nested `docs` folder atm, and I did specifically add a `docs/codebase-notes/README.md` that says "this material could go stale as we're refactoring, but I'm leaving it here so that it's saved somewhere and other folks can see it".